### PR TITLE
deletion-api: added end point for deletion of users in versioned router

### PR DESCRIPTION
**Pull request by ruchiramoitra**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/intercom/deleteUsers.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const { httpSend } = require("../../../adapters/network");
+const { CustomError } = require("../../util");
+
+const responseHandler = async (userAttributes, config) => {
```
Comment (saikumarrs) [inline v0/destinations/intercom/deleteUsers.js:4]: I believe this should be named something more appropriately, maybe `userDeletionHandler`.

File: versionedRouter.js [hunk 2]
```diff
@@ -640,6 +644,40 @@ const getJobStatus = async (ctx, type) => {
...
 };
 
+const handleDeletionOfUsers = async ctx => {
+  const { body } = ctx.request;
+  const respList = [];
+  let response;
+  await Promise.all(
+    body.map(async b => {
+      const { destType } = b;
+      const destDeletionHandler = getDeletionUserHandler("v0", destType);
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply versionedRouter.js:654]: or is this meant to be a generic deletion functionality and we are starting with 'users' now?

**Review by saikumarrs — COMMENTED** [review]
LGTM

**Review by saikumarrs — DISMISSED** [review]
LGTM

**Review by arnab-p — APPROVED** [review]
LGTM

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**